app/controllers/admin/court_applications_controller.rb

```diff
@@ -33,15 +33,22 @@ def create
 
     def show
       @court_application = CourtApplication.find(params[:id])
+      @court_application_hearings = @court_application.court_hearings
+
+      if @court_application.prosecution_case.present?
+        @prosecution_case = ProsecutionCase.find(@court_application.prosecution_case)
```

Comment:
This line is a bit weird, and I wonder if we can rewrite it to reduce the weirdness, or at least make what's weird about it more explicit.

`@court_application.prosecution_case` is _actually_, despite the name, a collection of prosecution cases (because CourtApplication `has_many :prosecution_case, through: :court_application_prosecution_case`)

So what this line does is take a collection of prosecution cases and pass them as arguments to ProsecutionCase.find. Which, to my surprise, doesn't explode, it just retrieves the first matching prosecution case.

So I think we could simplify this and make clearer what's going on by writing it as `@prosecution_case = @court_application.prosecution_case.first`

Reply:
Bonus point for fixing the grammar by changing the association to `has_many :prosecution_cases, through: :court_application_prosecution_cases)` etc

Reply:
This check is in place because the existing mock implementation does not have a case associated with it. This line is actually meant to prevent the following it from failing, so that if it is present, we can use it. I'm not sure if it will be clearer with your suggestion.

Reply:
> the existing mock implementation does not have a case associated with it.

The existing mock implementation has _multiple_ cases associated with it. All this line of code is doing is taking a collection of cases (in `@court_application.prosecution_case`) and picking the first one. I'm saying we might as well be explicit about that.

Reply:
Ohh... I think we're at cross purposes. Your comment is about the `if` statement, isn't it? Which I have no problem with. It's the next line, `@prosecution_case = ProsecutionCase.find(@court_application.prosecution_case)`, which I'm worried about. Since we've already instantiated the prosecution case and already have it available as the first element in `@court_application.prosecution_case`, going _back_ to the database and re-instantiating the case from scratch by doing a completely unnecessary `.find` is the thing that has me raising my eyebrows.